binn/explainer.py

```diff
@@ -98,21 +101,49 @@ def explain_average(self,
             df = self.explain(test_data, background_data)
             dfs[iteration] = df
 
-        col_names = [f'value_{n}' for n in range(len(list(dfs.keys())))]
+        col_names = [f"value_{n}" for n in range(len(list(dfs.keys())))]
         values = [df.value.values for df in dfs.values()]
         values = np.array(values)
         values_mean = np.mean(values, axis=0)
         values_std = np.std(values, axis=0)
         df = dfs[0].copy()
-        df.drop(columns=['value'], inplace=True)
+        df.drop(columns=["value"], inplace=True)
         df[col_names] = values.T
-        df['value_mean'] = values_mean
-        df['values_std'] = values_std
-        df['value'] = values_mean
+        df["value_mean"] = values_mean
+        df["values_std"] = values_std
+        df["value"] = values_mean
         return df
 
+    def recursive_pathway_elimination(
+        self,
+        input_data,
+        design_matrix,
+        nr_iterations: int = 20,
+        max_epochs: int = 50,
+        clip_threshold=1e-5,
+        constant_removal_rate=0.05,
+        min_features_per_layer=3,
+        early_stopping=True,
+    ):
+        rpe = RecursivePathwayElimination(self.model, self)
```

Comment:
Why pass in both self.model and self? This seems repetitive, is it necessary?

Reply:
True, can just pass self.